src/checkpoint/second/service/VehicleService.java

```diff
@@ -0,0 +1,138 @@
+package checkpoint.second.service;
+
+import checkpoint.second.model.Automaker;
+import checkpoint.second.model.Vehicle;
+import checkpoint.second.repository.VehicleRepository;
+
+import java.util.Objects;
+import java.util.Scanner;
+
+public class VehicleService {
+
+	VehicleRepository vehicleRepository = new VehicleRepository();
+
+	public Vehicle[] listAll() {
+		return vehicleRepository.listAll();
+	}
+
+	public Vehicle[] searchByAutomaker(Vehicle[] vehicles, String name) {
+		Vehicle[] newVehicles = new Vehicle[vehicles.length];
+		int i = 0;
+		for (Vehicle vehicle : vehicles) {
+			if (vehicle.getAutomaker().getName().equals(name)) {
+				newVehicles[i] = vehicle;
+				i++;
+			}
+		}
+		return newVehicles;
+	}
+
+	public Vehicle searchByModel(Vehicle[] vehicles, String model) {
+		Vehicle newVehicle = new Vehicle();
+		for (Vehicle vehicle : vehicles) {
+			if (vehicle.getModel().equalsIgnoreCase(model)) {
+				newVehicle.setModel(vehicle.getModel());
+				newVehicle.setColor(vehicle.getColor());
+				newVehicle.setYear(vehicle.getYear());
+				newVehicle.setAutomaker(vehicle.getAutomaker());
+			}
+		}
+		return newVehicle;
+	}
+
+	public void addVehicle(Vehicle[] vehicles, Vehicle vehicle) {
+		Vehicle checkByModel = searchByModel(vehicles, vehicle.getModel());
+		if (Objects.nonNull(checkByModel.getModel())) {
+			throw new IllegalArgumentException("This vehicle model already exists!");
+		}
+
+		Vehicle[] newVehicles = new Vehicle[vehicles.length + 1];
+		for (int i = 0; i < vehicles.length; i++) {
+			newVehicles[i] = vehicles[i];
+		}
+		newVehicles[vehicles.length] = vehicle;
+		vehicleRepository.setVehicles(newVehicles);
+	}
+
+	public void updateVehicle(Vehicle[] vehicles, Vehicle vehicle) {
```

Comment:
Quando for atualizar, você só precisa do veículo antigo e do novo, não precisa receber a lista como parâmetro